MapboxNavigation/InstructionsBannerViewLayout.swift

```diff
@@ -77,16 +83,24 @@ extension BaseInstructionsBannerView {
         // Primary Label
         primaryLabel.leadingAnchor.constraint(equalTo: dividerView.trailingAnchor).isActive = true
         primaryLabel.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -18).isActive = true
-        baselineConstraints.append(primaryLabel.topAnchor.constraint(equalTo: maneuverView.topAnchor))
+        baselineConstraints.append(primaryLabel.topAnchor.constraint(equalTo: maneuverView.topAnchor, constant: -BaseInstructionsBannerView.padding/2))
         centerYConstraints.append(primaryLabel.centerYAnchor.constraint(equalTo: centerYAnchor))
         
         // Secondary Label
         secondaryLabel.leadingAnchor.constraint(equalTo: dividerView.trailingAnchor).isActive = true
         secondaryLabel.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -18).isActive = true
-        baselineConstraints.append(secondaryLabel.lastBaselineAnchor.constraint(equalTo: distanceLabel.lastBaselineAnchor))
+        baselineConstraints.append(secondaryLabel.lastBaselineAnchor.constraint(equalTo: distanceLabel.lastBaselineAnchor, constant: -BaseInstructionsBannerView.padding/2))
         baselineConstraints.append(secondaryLabel.topAnchor.constraint(greaterThanOrEqualTo: primaryLabel.bottomAnchor, constant: 0))
         centerYConstraints.append(secondaryLabel.topAnchor.constraint(greaterThanOrEqualTo: primaryLabel.bottomAnchor, constant: 0))
         
+        // Drag Indicator View
+        stepListIndicatorView.heightAnchor.constraint(equalToConstant: BaseInstructionsBannerView.stepListIndicatorViewSize.height).isActive = true
+        stepListIndicatorView.widthAnchor.constraint(equalToConstant: BaseInstructionsBannerView.stepListIndicatorViewSize.width).isActive = true
+        stepListIndicatorView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -BaseInstructionsBannerView.padding/2).isActive = true
+        stepListIndicatorView.centerXAnchor.constraint(equalTo: centerXAnchor).isActive = true
+        baselineConstraints.append(stepListIndicatorView.topAnchor.constraint(greaterThanOrEqualTo: secondaryLabel.bottomAnchor))
+        centerYConstraints.append(stepListIndicatorView.topAnchor.constraint(greaterThanOrEqualTo: secondaryLabel.bottomAnchor, constant: 0))
```

Comment:
The [testPrimaryShieldAndSecondary@3x.png](https://github.com/mapbox/mapbox-navigation-ios/blob/392bb8925c07ff92d52e29bcc55828bab919e444/MapboxNavigationTests/ReferenceImages_64/MapboxNavigationTests.InstructionsBannerViewSnapshotTests/testPrimaryShieldAndSecondary%403x.png) is not aligned correctly.

Reply:
@frederoni what do you mean? And any suggestion on how to fix this?

Reply:
The distance label and the secondary instruction label should be aligned according to their baseline indicated by the horizontal red line:

<img width="621" alt="testprimaryshieldandsecondary_3x_misaligned" src="https://user-images.githubusercontent.com/764476/37896366-e885ebd4-30e3-11e8-855b-4ddbf724066b.png">

Reply:
That was intentional to accommodate the new drag bar introduced into the view layout. That's the most visually balanced layout that I could compose. The drag bar would have been truncated and pushed into the superview bottom margin otherwise.

Reply:
@frederoni ☝️